.github/actions/build-n-cache-image/action.yml

```
@@ -23,15 +43,49 @@ runs:
         - name: Checkout
           uses: actions/checkout@v4
 
+        - name: Set up Docker Buildx
+          uses: docker/setup-buildx-action@v3
+
+        - name: Set up QEMU
+          uses: docker/setup-qemu-action@v3
+
         - name: Set up Depot CLI
           uses: depot/setup-action@v1
 
+        - name: Configure AWS credentials
+          if: ${{ inputs.push-image == 'true' }}
+          uses: aws-actions/configure-aws-credentials@v4
+          with:
+              aws-access-key-id: ${{ inputs.aws-access-key }}
+              aws-secret-access-key: ${{ inputs.aws-access-secret }}
+              aws-region: us-east-1
+
+        - name: Login to DockerHub
+          if: ${{ inputs.push-image == 'true' }}
+          uses: docker/login-action@v3
+          with:
+              username: ${{ inputs.dockerhub-username }}
+              password: ${{ inputs.dockerhub-password }}
+
+        - name: Login to Amazon ECR
+          if: ${{ inputs.push-image == 'true' }}
+          id: aws-ecr
+          uses: aws-actions/amazon-ecr-login@v2
+
         - name: Emit image tag
           id: emit
           shell: bash
-          run: echo "tag=posthog/posthog:${{ github.sha }}" >> $GITHUB_OUTPUT
+          run: |
+              TAGS="posthog/posthog:${{ github.sha }}"
+              if [[ "${{ inputs.push-image }}" == "true" ]]; then
+                TAGS="$TAGS,${{ steps.aws-ecr.outputs.registry }}/posthog-cloud:commit-${{ github.sha }}"
```

```suggestion
                TAGS="$TAGS,${{ steps.aws-ecr.outputs.registry }}/posthog-cloud:pr-commit-${{ github.sha }}"
```

not sure, but `commit-` prefix sounds also like a legit image for prod

---

and we can write a simple ecr policy for cleaning everything with the `pr-` prefix